Formula/m/mender-artifact.rb

```diff
@@ -0,0 +1,41 @@
+class MenderArtifact < Formula
+  desc "CLI tool for managing Mender artifact files"
+  homepage "https://mender.io"
+  url "https://github.com/mendersoftware/mender-artifact.git", tag: "4.1.0", revision: "af3853ecb7aae71c535fb564b4145b98ff7265f6"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+  depends_on "pkgconf" => :build
+  depends_on "openssl@3"
+
+  def install
+    system("go",
+           "build",
+           *std_go_args(ldflags: "-s -w -X github.com/mendersoftware/mender-artifact/cli.Version=" + version))
+  end
+
+  test do
+    # Below we use the mender-artifact binary to create a Mender artifact and then read it to get
+    # its metadata.
+
+    # First, we need to create a payload file.
+    (testpath/"rootfs.ext4").write("")
+    system(bin/"mender-artifact",
+           "write", "rootfs-image",
+           "-t", "beaglebone",
+           "-n", "release-1",
+           "-f", "rootfs.ext4",
+           "-o", "artifact.mender")
+    assert_path_exists testpath/"artifact.mender"
+
+    output = shell_output(bin/"mender-artifact read artifact.mender 2>/dev/null").split("\n").map(&:strip)
+    [
+      "Mender Artifact:",
+      "Name: release-1",
+      "Format: mender",
+      "Version: 3",
+      "Compatible devices: [beaglebone]",
+      "- Type: rootfs-image",
+    ].each { |expected| assert_includes output, expected }
```

Comment:
improved test and also added a version check
```suggestion
    assert_match version.to_s, shell_output("#{bin}/mender-artifact --version")

    # Create a test artifact
    (testpath/"rootfs.ext4").write("")

    output = shell_output("#{bin}/mender-artifact write rootfs-image " \
                          "-t beaglebone -n release-1 -f rootfs.ext4 -o artifact.mender")
    assert_path_exists testpath/"artifact.mender"

    # Verify the artifact contents
    output = shell_output("#{bin}/mender-artifact read artifact.mender")
    assert_match <<~EOS, output
      Mender Artifact:
        Name: release-1
        Format: mender
        Version: 3
        Signature: no signature
        Compatible devices: [beaglebone]
    EOS
```